### DISCUSSION TITLE
Bus Transreceiver

### DISCUSSION THREAD

**Discussion by siddhc** [flandreas/antares#490] [Q&A]
I wanted to model the 74LS245 (Octal Bus Transreceiver). For this tried making one of the 8 modules like shown in the image. This is not working i.e. the Tristate buffers are triggering correctly but transmission and receiving is not happening. Is it because ordinary tristate buffers are being used (instead of schmitt triggers)? How can we use a schmitt trigger in Antares? Any other way of approaching?

Please let me know if you need any more information about the circuit.

![image](https://user-images.githubusercontent.com/18248762/196501113-e650162c-f1b0-49a6-b963-0678a1730328.png)

**Comment by flandreas** [comment]
@siddhc No, this should work with ordinary tristate buffers. That's pretty much the same circuit as contained in Antares' "Standard Library", see "Bus Transceiver 4 Bit" in the "Net" folder, and that one works as expected IMHO. Have you set EN to 0?

If you like, you can export your project and attach the .zip file to this issue, so I can look into it.

**Reply by siddhc** [reply]
[1 Channel Transreceiver.zip](https://github.com/flandreas/antares/files/9821247/1.Channel.Transreceiver.zip)
@flandreas Thank you for your reply. The .zip file is attached. Request your comments.

When I try the "Bus Transceiver 4 Bit" in the Antares' "Standard Library": 
Say EN=0 and DIR=0: Now B is tri-stated, so any input in B goes to A. But if I change the value at A, nothing happens at B. This is okay. But not I change B, nothing happens at A. It is normal? Request you to share.

**Comment by flandreas** [comment]
One pitfall also is that you have to make sure that the receiving inout doesn't assert a signal when sending a signal from the sending inout. If DIR is 0, signals go from B to A. If you click on A, either nothing will happen, or a conflict arises, resulting in red wires.

**Reply by siddhc** [reply]
@flandreas Yes this is happening. that is why it automatically resets to 0. Please suggest where I am going wrong. I also tried "Transmission Gate" in the "Net" directory, but could not work out a way.

**Comment by flandreas** [comment]
@siddhc Okay, I see the problem.

First of all, whenever using a bi-directional circuit like like, after sending a signal from B to A by clicking on A, you have to make sure to enter a Z in B before changing the signal flow direction. Otherwise it will result in a conflict when sending a signal from A to B, because the last asserted non-Z value will still stick at B. This works fine in "Bus Transceiver 4 Bit" from the Standard Library.

However, this doesn't seem to work in your circuit. There seems to a difference between 1-bit input/outputs and n-bit inputs/outputs in the way how the initial signal is treated by the input/output. This seems to be a bug. I will open a bug issue for that.

The bug seems to be restricted to clicking within the circuit. When I use your circuit as a subcircuit in another circuit, it works as expected provided that I enter Z in the input/output before changing DIR:

https://user-images.githubusercontent.com/24633141/196732488-ea8de7d8-b8a0-4dee-84dd-8cadaa94fdf7.mov

**Reply by siddhc** [reply]
@flandreas Thanks for your comments and the animation. Understood, will need a bit to get used to it.

**Comment by flandreas** [comment]
@siddhc While writing the bug issue, I've just discovered that the problem is that your A and B input/outputs don't have the property "Switch" set. Without that property, when pressing the mouse button on B, the resulting 1 signal is forwarded to A, but when releasing the mouse button, B doesn't go back to 0 because B is now determined by the 1 value of A.

If you set the "Switch" property on both A and B, everything works as expected, provided that you enter Z before changing DIR.
